clients/java/src/main/java/com/google/myanmartools/Phase.java

```diff
@@ -0,0 +1,123 @@
+/* Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.myanmartools;
+
+/**
+ * Class implementing a transliteration phase as an array of rules.
+ */
+
+import java.util.ArrayList;
+import java.util.regex.Matcher;
+
+public class Phase {
+
+  private final ArrayList<Rule> phaseRules;
+  private String info;
+  private Boolean debugMode;
+
+  public Phase() {
+    // Initialize the rules for the phase.
+    phaseRules = new ArrayList<>();
+    info = "";
+    debugMode = false;
+  }
+
+  public void setInfo(String newInfo) {
+    info = newInfo;
+  }
+
+  public void setDebugMode(Boolean newMode) {
+    debugMode = newMode;
+  }
+
+  public void addRule(Rule newRule) {
+    phaseRules.add(newRule);
+    // Put the rule number within the phase.
+    newRule.setInfo("" + phaseRules.size());
+  }
+
+  public String printPhase() {
+    StringBuilder sb = new StringBuilder();
+    sb.append("  Phase ").append(info)
+        .append(" has ").
+        append(phaseRules.size()).
+        append(" rules\n");
+
+    for (Rule rule : phaseRules) {
+      sb.append(rule.printRule());
+    }
+    return sb.toString();
+  }
+
+  public String runPhase(Phase phase, String inString) {
+    // Run all the rules of this phase.
+    String outString = "";
+    String midString = inString;
+    Boolean startOfString = true;
```

Comment:
Nit: Here and elsewhere, I recommend using the primitive type `boolean` (instead of the box type `Boolean`).  It will reduce object instantiations, which Android cares about.

Reply:
Done